### ISSUE TITLE
Question about lossAB usage and first cross-domain transition masking

### ISSUE THREAD

**Issue by tomatoboy-hub** [liuqidong07/LLM4CDSR-pytorch#3]
@liuqidong07 
Hi, I have a question about the training loss in the cross-domain sequential recommendation implementation.

In `forward()`, `lossAB` is computed from the mixed-domain sequence, but the final returned loss is:

```python
loss = lossA.mean() + lossB.mean()
```
So lossAB is not directly added.

For example, if the domain sequence is:

A, A, B, B, A, A, B, A

my understanding is that the first transition into domain B, i.e. A2 -> B1, is removed from domain_mask because there is no corresponding local-B transition for predicting B1.
However, later cross-domain transitions such as B2 -> A3, A4 -> B3, and B3 -> A5 are still used through the score addition, because each target domain already has local history.

Is this understanding correct?

Also, is there a specific reason why lossAB is computed but not added to the final loss?
Would adding lossAB cause duplicated supervision for transitions that are already trained through lossA / lossB after score addition?
